src/main/java/com/redhat/devtools/lsp4ij/LanguageServersRegistry.java

```diff
@@ -483,13 +486,15 @@ private void removeAssociationsFor(LanguageServerDefinition definition) {
         boolean mappingsChanged = !Objects.deepEquals(settings.getMappings(), request.mappings());
         boolean configurationContentChanged = !Objects.equals(settings.getConfigurationContent(), request.configurationContent());
         boolean initializationOptionsContentChanged = !Objects.equals(settings.getInitializationOptionsContent(), request.initializationOptionsContent());
+        // Not checking whether client config changed because that shouldn't result in a LanguageServerChangedEvent
```

Comment:
Specifically calling out that we don't need to include client config changes when computing whether or not to send an event here because the actual language server doesn't care at all about client-side config changes.

Reply:
For the moment I agree with you. Lets keep like this, but we could have a client settings which could impact the editor (like enable/disable lsp codelens which should refresh the editor and in this case we should send an event)